sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -1370,21 +1370,21 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
   def unsupportedOperandTypeForSizeFunctionError(
       dataType: DataType): SparkUnsupportedOperationException = {
     new SparkUnsupportedOperationException(
-      errorClass = "_LEGACY_ERROR_TEMP_2156",
+      errorClass = "UNSUPPORTED_DATA_TYPE_FOR_SIZE_FUNCTION",
       messageParameters = Map(
         "dataType" -> dataType.getClass.getCanonicalName))
   }
 
   def unexpectedValueForStartInFunctionError(prettyName: String): SparkRuntimeException = {
     new SparkRuntimeException(
-      errorClass = "_LEGACY_ERROR_TEMP_2157",
+      errorClass = "UNEXPECTED_VALUE_FOR_START_IN_SLICE_FUNCTION",
       messageParameters = Map(
         "prettyName" -> prettyName))
```

Comment:
Is the pretty name a id, if so, please, wrap it by `toSQLId`

Reply:
Good catch, I also replaced `dataType.getClass.getCanonicalName` above with `toSQLType(dataType)`